File: ProyectoMysql/server/BusinessLayer/MerLista.py
from DataLayer.MerListaDB import *
from Utilidades.Conexion.configMysql import StartTransaction, EndTransaction


class MerLista:

    def ObtenerMain(Codigo: str):
        try:
            data = MerListaDB.ObtenerMain(Codigo)
            return data
        except Exception as e:
            print(e)

    def Registrar(Ent: MerListaSaveModel):
        try:
            StartTransaction()
            data = MerListaDB.RegistrarDB(Ent)
            EndTransaction()
            return data
        except Exception as e:
            print(e)

    def ObtenerTitulo(Codigo: str):
        try:
            data = MerListaDB.ObtenerTitulo(Codigo)
            return data
        except Exception as e:
            print(e)

    def BuscarItem(Codigo: str, Nombre: str):
        try:
            data = MerListaDB.BuscarItem(Codigo, Nombre)
            return data
        except Exception as e:
            print(e)

    def ObtenerItem(Id : int):
        try:
            data = MerListaDB.ObtenerItem(Id)
            return data
        except Exception as e:
            print(e)

File: ProyectoMysql/Api/LogisticStorage/LogisticStorage.BusinessLayer/MerLista.cs
using LogisticStorage.DataLayer;
using LogisticStorage.EntityLayer;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LogisticStorage.BusinessLayer
{
    public class MerLista
    {
        public static List<MerListaEntity> ObtenerMain()
        {
            MerListaDB DB = new MerListaDB();
            return DB.ObtenerMain();
        }
        public static Int32 Registrar(MerListaEntity Ent)
        {
            MerListaDB DB = new MerListaDB();
            DB.Registrar(Ent);
            return Ent.ListaId;
        }
        public static List<MerListaEntity> ObtenerTitulo(String Codigo)
        {
            MerListaDB DB = new MerListaDB();
            return DB.ObtenerTitulo(Codigo);
        }

        public static List<MerListaEntity> BuscarItem(String Codigo, String Nombre)
        {
            MerListaDB DB = new MerListaDB();
            return DB.BuscarItem(Codigo, Nombre);
        }
        public static List<MerListaEntity> ObtenerItem(Int32 MercaderiaId)
        {
            MerListaDB DB = new MerListaDB();
            return DB.ObtenerItem(MercaderiaId);
        }
    }
}
